docker/efficiency/containers.py

```diff
@@ -0,0 +1,178 @@
+import io
+import tarfile
+
+import six
+
+from . import commons
+from . import images
+
+
+def copy_to_fs(client, container, path, target="."):
+    """
+    Copy file from container to filesystem
+
+    **Params:**
+        client: a docker `Client` object
+        container_id: ID of the container to copy from
+        path: path to the file in the container
+        target: folder where file will be copied (default ".")
+    """
+    response = client.copy(container, path)
+    buffer = io.BytesIO()
+    buffer.write(response.data)
+    buffer.seek(0)
+    tar = tarfile.open(fileobj=buffer, mode='r|')
+    tar.extractall(path=target)
+
+
+def start_auto_remove(client, container, *args, **kwargs):
+    """
+    Start a container and try to remove it when it's finished running,
+    similar to using --autorm in the docker CLI.
+    **Params:**
+        client: a docker `Client` object
+        container: ID of the container to be started
+        args, kwargs: `Client.start()` arguments
+    """
+    client.start(container, *args, **kwargs)
+    if client.wait(container) == 0:
+        return client.remove_container(container)
+
+
+class Exec(commons.Interactive):
+    def __init__(self, client, id):
+        super(Exec, self).__init__(client, id)
+        data = self._client.exec_inspect(self.id)
+        for k, v in six.iteritems(data):
+            if k == 'Id':
+                continue
+            setattr(self, commons.to_snakecase(k), v)
+
+    def start(self, detach=False, tty=False, stream=False):
+        return self._client.exec_start(self.id, detach, tty, stream)
+
+    def resize(self, height=None, width=None):
+        self._client.exec_resize(self.id, height, width)
+
+    def __str__(self):
+        return '<DockerExec(id={0})>'.format(self.id[:16])
+
+    def __repr__(self):
+        return str(self)
+
+
+class Container(commons.Interactive):
```

Comment:
I think we should to make this class compatible with `compose.container.Container` so that we can replace it with this one.